src/lib/io/sol-gpio-common.c

```diff
@@ -74,3 +77,34 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
 
     return gpio;
 }
+
+SOL_API enum sol_gpio_edge
+sol_gpio_edge_from_str(const char *edge)
+{
+    static const struct sol_str_table table[] = {
```

Comment:
On Tue, 05 Apr 2016 00:08:44 -0700, gabrielschulhof wrote:
> > @@ -74,3 +77,34 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
> >  
> >      return gpio;
> >  }
> > +
> > +SOL_API enum sol_gpio_edge
> > +sol_gpio_edge_from_str(const char *edge)
> > +{
> > +    static const struct sol_str_table table[] = {
> 
> @cabelitos will it crash even though```sol_str_slice_from_str()``` is declared ```SOL_ATTR_NONNULL(1)```?
> 

The nonnull attribute tells the compiler that the argument will never be
NULL, so if you pass an explicit NULL there, it will complain at build
time, and for other cases, it's free to optimize away any checks inside
the function.
That passes the responsibility to the user of the function to ensure
that the parameter passed will never be NULL.